### PR TITLE
Adds guard around terminator bytes casting. 

### PR DESCRIPTION
**Pull request by sabedevops**
Lint fix and typing improvements.

### CODE REVIEW

File: src/openziti/zitisock.py [hunk 1]
```diff
@@ -106,7 +105,9 @@ def create_ziti_connection(address, **_):
     sock.connect(address)
     return sock
 
+
 def ziti_getaddrinfo(host, port, family=0, type=0, proto=0, flags=0):
+    # pylint: disable=redefined-builtin
```
Comment (ekoby) [inline src/openziti/zitisock.py:110]: why is this needed?

**Review by ekoby — APPROVED** [review]

File: src/openziti/zitisock.py (hunk 1 above)
Comment (sabedevops) [reply src/openziti/zitisock.py:110]: `type` is a builtin: https://docs.python.org/3/library/functions.html#type

In this case, I assumed this came from the conventions used by the `socket` module, so I silenced the message.
